include/swift/SIL/SILInstruction.h

```diff
@@ -8094,6 +8094,11 @@ inline EnumElementDecl **SelectEnumInstBase::getEnumElementDeclStorage() {
   llvm_unreachable("Unhandled SelectEnumInstBase subclass");
 }
 
+inline void SILSuccessor::pred_iterator::cacheBasicBlock() {
+  // No need to assert(Cur). Dereferencing nullptr always crashes.
```

Comment:
While it may crash, providing a nice error message when we hit that condition can be useful when triaging bugs and also provides a nice guide for the reader of the code that a pointer is intended to be non-null. Can you put that back in?